include/boost/geometry/io/wkt/write.hpp

```diff
@@ -62,22 +64,94 @@ namespace boost { namespace geometry
 namespace detail { namespace wkt
 {
 
+template <typename Output>
+struct output_formatter
+{
+    template <typename T>
+    static void append(Output& out, T const& v, int = -1)
+    {
+        out << v;
+    }
+};
+
+template <>
+struct output_formatter<std::string>
+{
+    template <typename T>
+    static void append(std::string& out, T const& val, int significant_digits = -1)
+    {
+        BOOST_STATIC_ASSERT((boost::is_arithmetic<T>::value));
+        std::stringstream ss;
+        if(std::numeric_limits<T>::is_specialized)
+        {
+            if(significant_digits < 0)
+                ss.precision(std::numeric_limits<T>::digits10);
+            else
+                ss.precision(significant_digits);
+        }
+        ss << val;
+        out += ss.str();
+    }
+
+    static void append(std::string& out, char c, int = -1)
+    {
+        out += c;
+    }
+
+    static void append(std::string& out, char const* s, int = -1)
+    {
+        out += s;
+    }
+};
+
+template <>
+struct output_formatter<std::ostream>
+{
+    template <typename T>
+    static void append(std::ostream& out, T const& val, int significant_digits = -1)
+    {
+        BOOST_STATIC_ASSERT((boost::is_arithmetic<T>::value));
+        if(std::numeric_limits<T>::is_specialized)
```

Comment:
Setting precision is only needed for string output isn't it? Why is it here? Did I miss something?

Reply:
No, you didn't miss anything.

It's just that both, `wkt` stream manipulator and `to_wkt` function use the same formatter implementation, we just got carried away and applied the dedicated argument a bit too far indeed.

Reply:
Ok, so `significant_digits` is used here, and in specialization for `std::string`, but not in the default/unspecialized `output_formatter<Output>` (I'm not saying that it should be, I think it should only be used for `std::string`).

AFAIU this specialization for `std::ostream` was added exactly to support `significant_digits` for streams but it will not work for all of them, only for `std::basic_ostream<char>` as long as `std::ostream` is a typedef of `std::basic_ostream<char>` not a separate class because then the default/unspecialized `output_formatter<Output>` would be used.

Is it guaranteed by the standard that `std::ostream` is a typedef and not a separate class?

So anyway, assuming it is, this will be done only for streams using `char`.

I'd just remove the whole specialization together with `significant_digits` from `bg::wkt` stream manipulator. Is my reasoning ok?

Reply:
> I'd just remove the whole specialization together with significant_digits from bg::wkt stream manipulator. Is my reasoning ok?

@awulkiew Yes, as I explained in https://github.com/boostorg/geometry/pull/670#discussion_r394002323, you've made good catch of this unnecessary change in the `wkt` manipulator.

@sudo-panda I'm sorry I had not caught that earlier myself. I somehow concentrated on the `to_wkt` addition and overlooked how it affects the existing manipulator.

Reply:
@mloskot no need to say sorry. I guess sometimes you need someone not involved in the process to look at the bigger picture. I am here to learn no matter how long it takes. 
@awulkiew  thanks for the review. I will push all these changes soon.